src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1336,6 +1455,73 @@ public IEnumerable<(string Key, long Index)> GetCompareExchangeIndexes(Transacti
             }
         }
 
+        public IEnumerable<(string Key, long Index, BlittableJsonReaderObject Value)> GetCompareExchangeFrom(TransactionOperationContext context, string dbName, long fromIndex, int take)
+        {
+            CompareExchangeCommandBase.GetPrefixIndexSlices(context.Allocator, dbName, fromIndex, out var indexTuple);
+
+            using (indexTuple.Scope)
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(CompareExchangeSchema, CompareExchange);
+                using (Slice.External(context.Allocator, indexTuple.Buffer, indexTuple.Buffer.Length, out var keySlice))
+                    foreach (var tvr in table.SeekForwardFrom(CompareExchangeSchema.Indexes[CompareExchangeIndexSlice], keySlice, 0))
```

Comment:
Is it possible to read values for a different database here?

Reply:
yes it returns all values starting from keyslice.

Reply:
You need to make sure that this doesn't happen, otherwise, we might leak data from another db, etc.